### PR TITLE
make Extract additional trash cost optional

### PR DESCRIPTION
**Pull request by francescopellegrini**
Reworked to resemble other _may_ abilities.

### CODE REVIEW

**Review by NoahTheDuke — CHANGES_REQUESTED** [review]
If we're moving these implicit optional to explicit, they should have reqs in the :optional map to make sure they can be used.

**Comment by francescopellegrini** [comment]
> If we're moving these implicit optional to explicit, they should have reqs in the :optional map to make sure they can be used.

In the end I refactored it to resemble Lateral Growth. I think it's way more readable now

File: src/clj/game/cards/operations.clj [hunk 1]
```diff
@@ -1322,13 +1319,15 @@
     :effect (req (wait-for (gain-credits state side 4)
                            (continue-ability
                              state side
-                             {:player :corp
-                              :prompt "Choose a card to install"
+                             {:prompt "Choose a card to install"
+                              :req (req (not-empty (filter #(corp-installable-type? %) (:hand corp))))
```
Comment (NoahTheDuke) [inline src/clj/game/cards/operations.clj:1323]:
```suggestion
                              :req (req (not-empty (filter corp-installable-type? (:hand corp))))
```

File: src/clj/game/cards/operations.clj [hunk 2]
```diff
@@ -756,24 +756,21 @@
...
-                                              (gain-credits state side eid 3)))}
-                      card nil))
-                   ;; no cards to trash -> skip prompt, no info is given away
-                   (do (system-msg state side "uses extract to gain 6 [credit]")
-                       (gain-credits state side eid 6))))}})
+    :msg "gain 6 [Credit]"
+    :effect (req (wait-for (gain-credits state side 6)
+                           (continue-ability
+                             state side
+                             {:prompt "Choose an installed card to trash"
```
Comment (NoahTheDuke) [inline src/clj/game/cards/operations.clj:763]: Maybe add a `:waiting-prompt` too?

**Review by NoahTheDuke — CHANGES_REQUESTED** [review]

**Pull request merged**